### PR TITLE
Add broadcast_callback()

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Make Issue #896 possible

### CODE REVIEW

File: src/taipy/gui/gui_actions.py [hunk 1]
```diff
@@ -243,6 +243,29 @@ def invoke_callback(
...
 
+def invoke_shared_callback(
+    gui: Gui,
+    callback: t.Callable,
+    args: t.Union[t.Tuple, t.List],
+    module_context: t.Optional[str] = None,
+) -> t.Any:
+    """Invoke a user callback using state with shared variables.
+
+    See the [User Manual section on Long Running Callbacks in a Thread](../../gui/callbacks/#long-running-callbacks-in-a-thread)
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/gui_actions.py]: long running ?

File: src/taipy/gui/gui_actions.py [hunk 2]
```diff
@@ -243,6 +243,29 @@ def invoke_callback(
...
+    Arguments:
+        gui (Gui^): The current gui instance.
+        callback (Callable[[State^, ...], None]): The user-defined function that is invoked.<br/>
+            The first parameter of this function **must** be a `State^`.
+        args (Union[Tuple, List]): The remaining arguments, as a List or a Tuple.
+        module_context (Optional[str]): the name of the module that will be used.
+    """
+    if isinstance(gui, Gui):
+        return gui._call_shared_user_callback(callback, list(args), module_context)
+    _warn("'invoke_callback()' must be called with a valid Gui instance.")
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/gui_actions.py]: invoke_shared...

**Review by FredLL-Avaiga — APPROVED** [review]
a bit scary but you  know best

File: src/taipy/gui/gui_actions.py (hunk 2 above)
Comment (dinhlongviolin1) [reply src/taipy/gui/gui_actions.py]: thanks!

File: src/taipy/gui/gui_actions.py (hunk 1 above)
Comment (dinhlongviolin1) [reply src/taipy/gui/gui_actions.py]: i copied from the above gui action. not sure if it is correct though

Comment (dinhlongviolin1) [reply src/taipy/gui/gui_actions.py]: @FabienLelaquais can verify I guess

File: src/taipy/gui/data/data_scope.py [hunk 3]
```diff
@@ -27,7 +27,7 @@ def __init__(self) -> None:
     def set_single_client(self, value: bool) -> None:
         self.__single_client = value
 
-    def get_single_client(self) -> bool:
+    def is_single_client(self) -> bool:
```
Comment (FabienLelaquais) [inline src/taipy/gui/data/data_scope.py:30]: 👍

File: src/taipy/gui/gui_actions.py [hunk 4]
```diff
@@ -243,6 +243,29 @@ def invoke_callback(
     _warn("'invoke_callback()' must be called with a valid Gui instance.")
 
 
+def invoke_shared_callback(
```
Comment (FabienLelaquais) [inline src/taipy/gui/gui_actions.py]: What about 'broadcast_callback(gui: Gui, callback: Callable, *args)'?

File: src/taipy/gui/gui_actions.py (hunk 1 above)
Comment (FabienLelaquais) [reply src/taipy/gui/gui_actions.py]: Yes. I would actually replace the doc by "TODO".
So we don't forget.

Fred points out that the current content refers to the long_callback thing. Which is wrong.

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
Let's discuss the late adjustments ok?

File: src/taipy/gui/gui.py [hunk 5]
```diff
@@ -1125,6 +1127,30 @@ def _call_user_callback(
...
+                setattr(g, Gui.__SHARED_CALLBACK_G_ID, True)
+                callback_result = self._call_function_with_state(user_callback, args)
+                setattr(g, Gui.__SHARED_CALLBACK_G_ID, False)
+                return callback_result
+        except Exception as e:  # pragma: no cover
+            if not self._call_on_exception(user_callback.__name__, e):
+                _warn(f"invoke_callback(): Exception raised in '{user_callback.__name__}()':\n{e}")
+        return None
+
+    def _is_in_shared_callback(self):
```
Comment (FabienLelaquais) [inline src/taipy/gui/gui.py]: ...

File: src/taipy/gui/gui_actions.py (hunk 4 above)
Comment (FabienLelaquais) [reply src/taipy/gui/gui_actions.py]: That does not work, ok.

**Review by FabienLelaquais — APPROVED** [review]
👍

**Pull request merged**